torch/utils/cpp_extension.py

```diff
@@ -254,13 +262,16 @@ def unix_wrap_compile(obj, src, ext, cc_args, extra_postargs, pp_opts):
             try:
                 original_compiler = self.compiler.compiler_so
                 if _is_cuda_file(src):
-                    nvcc = _join_cuda_home('bin', 'nvcc')
+                    nvcc = (_join_cuda_home('bin', 'hipcc') if HIP_COMP else _join_cuda_home('bin', 'nvcc'))
                     if not isinstance(nvcc, list):
                         nvcc = [nvcc]
                     self.compiler.set_executable('compiler_so', nvcc)
                     if isinstance(cflags, dict):
                         cflags = cflags['nvcc']
-                    cflags = COMMON_NVCC_FLAGS + ['--compiler-options', "'-fPIC'"] + cflags
+                    if HIP_COMP:
+                       cflags = COMMON_HIPCC_FLAGS + cflags
+                    else:
+                       cflags = COMMON_NVCC_FLAGS + ['--compiler-options', "'-fPIC'"] + cflags
```

Comment:
There is a new merge conflict here from master adding `+ _get_cuda_arch_flags(cflags)`.